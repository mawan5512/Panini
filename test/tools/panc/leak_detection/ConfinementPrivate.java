/*
 * This file is part of the Panini project at Iowa State University.
 *
 * The contents of this file are subject to the Mozilla Public License
 * Version 1.1 (the "License"); you may not use this file except in
 * compliance with the License. You may obtain a copy of the License at
 * http://www.mozilla.org/MPL/.
 *
 * Software distributed under the License is distributed on an "AS IS" basis,
 * WITHOUT WARRANTY OF ANY KIND, either express or implied. See the License
 * for the specific language governing rights and limitations under the
 * License.
 *
 * For more details and the latest version of this code please see
 * http://paninij.org
 *
 * Contributor(s): Hridesh Rajan
 */

/*
 * @test
 * @summary Ensure Capsule state will not be leaked to other Capsule.
 * @compile/ref=ConfinementPrivate.out -XDrawDiagnostics  ConfinementPrivate.java
 */

class Data { }
capsule C {
	void cM(Data d) { }
}
capsule ConfinementPrivate {
	design {
		C c; 
	}
	Data d = new Data(); 
	void run() {
		privateM();
	}
	private void privateM(){
		c.cM(d); 
	}
}
