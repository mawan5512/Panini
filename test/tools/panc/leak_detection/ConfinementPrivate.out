ConfinementPrivate.java:39:22: compiler.warn.confinement.violation: d, ConfinementPrivate, privateM()
1 warning
